# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1518545547 -3600
#      Tue Feb 13 19:12:27 2018 +0100
# Node ID 4aa86466f97163f89ada19b73e5051e4a03ee36f
# Parent  6114be5172407a8a514638ad60015df8a358d490
fix font combobox in settings for variable editor (bug #53136)

* settings-dialog.cc (settings_dialog): initialize font name instead of font
  font style, en/disable combobox depending on checkbox for using terminal font

diff --git a/libgui/src/settings-dialog.cc b/libgui/src/settings-dialog.cc
--- a/libgui/src/settings-dialog.cc
+++ b/libgui/src/settings-dialog.cc
@@ -568,19 +568,27 @@ settings_dialog::settings_dialog (QWidge
   read_terminal_colors (settings);
 
   // variable editor
   ui->varedit_columnWidth->setText (settings->value ("variable_editor/column_width", "100").toString ());
   ui->varedit_autoFitColumnWidth->setChecked (settings->value ("variable_editor/autofit_column_width", false).toBool ());
   ui->varedit_autofitType->setCurrentIndex (settings->value ("autofit_type", 0).toInt ());
   ui->varedit_rowHeight->setText (settings->value ("variable_editor/row_height", "2").toString ());
   ui->varedit_rowAutofit->setChecked (settings->value ("variable_editor/autofit_row_height", true).toBool ());
-  ui->varedit_font->setFont (QFont (settings->value ("variable_editor/font_name", settings->value ("terminal/fontName", "Courier New")).toString ()));
+
+  ui->varedit_font->setCurrentFont (QFont (settings->value ("variable_editor/font_name", settings->value ("terminal/fontName", "Courier New")).toString ()));
   ui->varedit_fontSize->setValue (settings->value ("variable_editor/font_size", QVariant (10)).toInt ());
+  connect (ui->varedit_useTerminalFont, SIGNAL (toggled (bool)),
+           ui->varedit_font, SLOT (setDisabled (bool)));
+  connect (ui->varedit_useTerminalFont, SIGNAL (toggled (bool)),
+           ui->varedit_fontSize, SLOT (setDisabled (bool)));
   ui->varedit_useTerminalFont->setChecked (settings->value ("variable_editor/use_terminal_font", false).toBool ());
+  ui->varedit_font->setDisabled (ui->varedit_useTerminalFont->isChecked ());
+  ui->varedit_fontSize->setDisabled (ui->varedit_useTerminalFont->isChecked ());
+
   ui->varedit_alternate->setChecked (settings->value ("variable_editor/alternate_rows", QVariant (false)).toBool ());
 
   // variable editor colors
   read_varedit_colors (settings);
 
   // shortcuts
 
   ui->cb_prevent_readline_conflicts->setChecked (settings->value ("shortcuts/prevent_readline_conflicts", true).toBool ());
